File: ThreadsApp/ThreadsApp/View/Chat/Cells/Select/SelectCell.m
//
//  SelectCell.m
//  ThreadsApp
//
//  Created by Tabriz Dzhavadov on 13/07/2017.
//  Copyright © 2017 Sequenia. All rights reserved.
//

#import "SelectCell.h"
#import "SelectViewController.h"

@interface SelectCell()

@property (weak, nonatomic) IBOutlet UILabel *placeholderLabel;

@property (weak, nonatomic) IBOutlet UILabel *mainLabel;

@end

@implementation SelectCell

- (void) awakeFromNib {
    [super awakeFromNib];
}

- (void) construct {
    self.placeholderLabel.text = [CellTypeHelper textByType: self.type];
    self.mainLabel.text = [[self class] designName: self.selectedDesign];
}

- (void) showDesignControllerIn: (UIViewController *) controller
                        designs: (NSArray <NSNumber *> *) designs {
    SelectViewController *selectVC = [SelectViewController new];
    selectVC.selects = designs;
    selectVC.delegate = self.delegate;
    
    UINavigationController *nav = [UINavigationController new];
    nav.viewControllers = @[selectVC];
    
    [controller presentViewController: nav animated: YES completion: nil];
}

+ (NSString *) designName: (THRDesign) design {
    if (design == THRDesignBRS)
        return @"BRS";
    else
        return @"Default";
}

+ (NSArray <NSNumber *> *) allDesigns {
    NSMutableArray <NSNumber *> *designs = [@[] mutableCopy];
    for (THRDesign design = 0; design < THRDesignCount; design++) {
        [designs addObject: @(design)];
    }
    return designs.copy;
}

@end

File: ThreadsApp/ThreadsAppSwift/View/Chat/Cells/SelectCell.swift
//
//  SelectCell.swift
//  THRAppSwift-Polarbear
//
//  Created by Brooma Service on 16/05/2019.
//  Copyright © 2019 Brooma Service. All rights reserved.
//

import Foundation
import UIKit

enum THRDesign : UInt, CaseIterable {
    case Default
    case BRS
}

protocol SelectCellDelegate : class {
    func selectCellDidSelect(design: THRDesign)
}

class SelectCell : UITableViewCell {
    
    var type : CellType = .Design
    weak var delegate : SelectCellDelegate?
    var selectedDesign : THRDesign = .Default
    
    @IBOutlet weak var placeholderLabel : UILabel!
    @IBOutlet weak var mainLabel : UILabel!
    
    func construct() {
        self.placeholderLabel.text = CellTypeHelper.getCellText(forType: self.type)
        self.mainLabel.text = SelectCell.designName(design: selectedDesign)
    }
    
    func showDesignController(in controller: UIViewController, withDesigns designs: Array<THRDesign>) {
        let selectVC = SelectViewController()
        selectVC.selects = designs
        selectVC.delegate = self.delegate
        
        let navController = UINavigationController()
        navController.viewControllers = [selectVC]
        
        controller.present(navController, animated: true, completion: nil)
    }
    
    class func designName(design: THRDesign) -> String {
        
        switch design {
        case .BRS:
            return "BRS"
        case .Default:
            return "Default"
        }
    }
}
